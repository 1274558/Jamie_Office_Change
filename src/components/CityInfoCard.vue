<template>
  <v-card class="mx-auto">
    <v-img class="white--text align-end" 
      height="200px" 
      :src="`${cityData.city}.jpg`"
    >
      <v-card-title>{{cityData.city}}</v-card-title>
    </v-img>
    <v-card-subtitle class="pb-0">{{cityData.country}}</v-card-subtitle>
    <v-card-text class="text--primary">
      <v-row
        v-if="cityData.weather" 
        align="center">
        <v-col id="minimum-temperature">
          <i class="fas fa-thermometer-empty temp-font"></i>
          Min: {{cityData.weather.Temperature.Minimum.Value}}&deg;{{cityData.weather.Temperature.Minimum.Unit}}
        </v-col>
        <v-col id="maximum-temperature">
          <i class="fas fa-thermometer-full temp-font"></i>
          Max: {{cityData.weather.Temperature.Maximum.Value}}&deg;{{cityData.weather.Temperature.Minimum.Unit}}
        </v-col>
      </v-row>
      <v-row align="center">
        <v-col>
          <v-tabs 
            v-model="cityData.selectedTab" 
            icons-and-text>
            <v-tab>
              Day
              <v-icon>wb_sunny</v-icon>
            </v-tab>
            <v-tab>
              Night
              <v-icon>nights_stay</v-icon>
            </v-tab>
            <v-tab>
              Flight
              <v-icon>airplanemode_active</v-icon>
            </v-tab>
            <v-tabs-items v-model="cityData.selectedTab">
              <v-tab-item>
                 <div v-if="!cityData.weather">
                  <v-skeleton-loader
                    ref="skeleton"
                    :loading="loading"
                    :type="type"
                    class="mx-auto"
                  >
                  </v-skeleton-loader>
                </div>
                <div v-if="cityData.weather">
                  <v-row>
                    <v-col cols="6">
                      <v-img 
                        id="weather-day-con" 
                        :src="`/accuweathericons/${cityData.weather.Day.Icon}-s.png`"
                        >
                      </v-img>
                    </v-col>
                    <v-col cols="6">
                      <v-icon>toys</v-icon>
                      <label id="weather-day-wind-speed">Wind : {{cityData.weather.Day.Wind.Speed.Value}} {{cityData.weather.Day.Wind.Speed.Unit}}</label>
                      <hr />
                      <v-icon>invert_colors</v-icon>
                      <label id="weather-day-rain-speed">Rain : {{cityData.weather.Day.Rain.Value}}/ {{cityData.weather.Day.Rain.Unit}}</label>
                      <hr />
                      <v-icon>ac_unit</v-icon>
                      <label id="weather-day-snow">Snow : {{cityData.weather.Day.Snow.Value}}/ {{cityData.weather.Day.Snow.Unit}}</label>
                      <hr />
                    </v-col>
                  </v-row>
                </div>
                <div v-else>
                  <v-row class="no-data">
                    <label>{{noDataAvailable}}</label>
                  </v-row>
                </div>
                <v-row v-if="cityData.weather">
                  <v-col>
                    <h2>{{cityData.weather.Day.ShortPhrase}}</h2>
                    <span>{{cityData.weather.Day.LongPhrase}}</span>
                  </v-col>
                </v-row>
              </v-tab-item>
              <v-tab-item>
                <div v-if="!cityData.weather">
                  <v-skeleton-loader
                    ref="skeleton"
                    :loading="loading"
                    :type="type"
                    class="mx-auto"
                  >
                  </v-skeleton-loader>
                </div>
                <div v-if="cityData.weather">
                <v-row>
                  <v-col cols="6">
                    <v-img 
                      :src="`/accuweathericons/${cityData.weather.Night.Icon}-s.png`"
                    >
                    </v-img>
                  </v-col>
                  <v-col cols="6">
                    <v-icon>toys</v-icon>
                    <label id="weather-night-wind-speed">Wind : {{cityData.weather.Night.Wind.Speed.Value}} {{cityData.weather.Night.Wind.Speed.Unit}}</label>
                    <hr />
                    <v-icon>invert_colors</v-icon>
                    <label id="weather-night-rain-speed">Rain : {{cityData.weather.Night.Rain.Value}}/ {{cityData.weather.Night.Rain.Unit}}</label>
                    <hr />
                    <v-icon>ac_unit</v-icon>
                    <label id="weather-night-snow">Snow : {{cityData.weather.Night.Snow.Value}}/ {{cityData.weather.Night.Snow.Unit}}</label>
                    <hr />
                  </v-col>
                </v-row>
                <v-row v-if="cityData.weather">
                  <v-col>
                    <h2>{{cityData.weather.Night.ShortPhrase}}</h2>
                    <span>{{cityData.weather.Night.LongPhrase}}</span>
                  </v-col>
                </v-row>
                </div>
                <div v-else>
                  <v-row class="no-data">
                    <label>{{noDataAvailable}}</label>
                  </v-row>
                </div>
              </v-tab-item>
              <v-tab-item>
                <div v-if="!cityData.flights">
                  <v-skeleton-loader
                    ref="skeleton"
                    :loading="loading"
                    :type="type"
                    class="mx-auto"
                  >
                  </v-skeleton-loader>
                </div>
                <div v-if="cityData.flights && (!cityData.error) && cityData.flights.length">
                  <v-row v-for="flight in cityData.flights" :key="flight.id">
                    <v-col cols="4">
                      <v-icon>airplanemode_active</v-icon>
                    </v-col>
                    <v-col cols="8">
                      <label id="flight-source-destination">{{flight.cityFrom}} - {{flight.cityTo}}</label>
                      <hr />
                      <label id="flight-price">Price : {{flight.currency}} {{flight.price}}</label>
                      <hr />
                      <label id="flight-distance">Distance : {{flight.distance}}</label>
                      <hr />
                      <label id="flight-duration">Duration : {{flight.fly_duration}}</label>
                      <hr />
                      <label id="flight-stops">Stops : {{flight.technical_stops}}</label>
                      <hr />
                      <label id="flight-seats">Seats available : {{flight.availability.seats}}</label>
                      <hr />
                    </v-col>
                  </v-row>
                </div>
                <div v-else-if="cityData.error">
                  <v-row class="no-data">
                    <label>{{ noFlightDetails }}</label>
                  </v-row>
                </div>
                <div v-else>
                  <v-row class="no-data">
                    <label>{{ noDataAvailable }}</label>
                  </v-row>
                </div>
              </v-tab-item>
              <v-row>
                <v-btn 
                  type="submit" 
                  color="warning" 
                  @click="$emit('choose-city', cityData.city)"
                  >
                  {{cityToSelectButtonText}}
                </v-btn>
              </v-row>
            </v-tabs-items>
          </v-tabs>
        </v-col>
      </v-row>
    </v-card-text>
  </v-card>
</template>

<script>
  export default {
    name: 'CityInfoCard',
    props: {
      cityData: Object
    },
    data(){
      return {
        loading: true,
        type: 'list-item-avatar-three-line',
        noDataAvailable: 'No data available',
        noFlightDetails: 'You cannot enter the same city that you are in.',
        cityToSelectButtonText: 'Select',
        selectedTab: null,
      }
    }
  }
</script>

<style scoped>
  .temp-font {
    font-size: 2rem;
  }
  .no-data {
    font-size: 20px;
    padding: 15px;
    justify-content: center;
  }
</style>