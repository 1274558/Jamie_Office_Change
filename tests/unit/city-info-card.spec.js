import Vuetify from 'vuetify';
import Vue from "vue"
import CityInfoCard from "@/components/CityInfoCard";
import { createLocalVue, mount } from "@vue/test-utils";

const localVue = createLocalVue()

describe("CityInfoCard.vue", () => {
    let wrapper;
    let vuetify;

    beforeEach(() => {
        vuetify = new Vuetify();
        Vue.use(vuetify);
        wrapper = mount(CityInfoCard, {
            localVue,
            vuetify,
            propsData: {
                cityData: {
                    city: "Amsterdam",
                    countryCode: "NL",
                    airportCode: 'AMS',
                    country: "Netherlands",
                    weather: {
                        Temperature: {
                            Minimum: {
                                Value: 21.1
                            },
                            Maximum: {
                                Value: 32.2
                            }
                        },
                        Day: {
                            Icon: 6,
                            Wind: {
                                Speed: {
                                    Value: 18.8
                                }
                            },
                            Rain: {
                                Value: 23.1
                            },
                            Snow: {
                                Value: 10.1
                            }
                        },
                        Night: {
                            Icon: 6,
                            Wind: {
                                Speed: {
                                    Value: 15.0
                                }
                            },
                            Rain: {
                                Value: 56.0
                            },
                            Snow: {
                                Value: 10.2
                            }
                        }
                    },
                    flights: [{
                        cityFrom: "India",
                        cityTo: "Amsterdam",
                        price: 490,
                        distance: 6584.75,
                        fly_duration:"43h 20m",
                        technical_stops: 0,
                        availability:
                            {
                                seats: 7
                            }
                    }],
                    currency: 'EUR',
                  }
              }
        });
    })

    it('renders cityinfo card', async () => {
        await Vue.nextTick();
        expect(wrapper.exists()).toBe(true);
    });
    it('should display temperature details', async () => {
        await Vue.nextTick();
        expect(wrapper.find('#minimum-temperature').text()).toBe("Min: 21.1°");
        expect(wrapper.find('#maximum-temperature').text()).toBe("Max: 32.2°");
    });
    it('should display day weather details', async () => {
        await Vue.nextTick();
        expect(wrapper.find('#weather-day-wind-speed').text()).toBe("Wind : 18.8");
        expect(wrapper.find('#weather-day-rain-speed').text()).toBe("Rain : 23.1/");
        expect(wrapper.find('#weather-day-snow').text()).toBe("Snow : 10.1/");
    });
    it('should display night weather details', async () => {
        await Vue.nextTick();
        expect(wrapper.find('#weather-night-wind-speed').text()).toBe("Wind : 15");
        expect(wrapper.find('#weather-night-rain-speed').text()).toBe("Rain : 56/");
        expect(wrapper.find('#weather-night-snow').text()).toBe("Snow : 10.2/");
    });
    it('should display flight details', async () => {
        await Vue.nextTick();
        expect(wrapper.find('#flight-source-destination').text()).toBe("India - Amsterdam");
        expect(wrapper.find('#flight-price').text()).toBe("Price : EUR 490");
        expect(wrapper.find('#flight-distance').text()).toBe("Distance : 6584.75");
        expect(wrapper.find('#flight-duration').text()).toBe("Duration : 43h 20m");
        expect(wrapper.find('#flight-stops').text()).toBe("Stops : 0");
        expect(wrapper.find('#flight-seats').text()).toBe("Seats available : 7");
    });
})