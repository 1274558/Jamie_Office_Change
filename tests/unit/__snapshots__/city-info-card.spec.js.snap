// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityInfoCard.vue renders cityinfo card 1`] = `
<v-card class="mx-auto">
  <v-img height="200px" src="Amsterdam.jpg" class="white--text align-end">
    <v-card-title>Amsterdam</v-card-title>
  </v-img>
  <v-card-subtitle class="pb-0">Netherlands</v-card-subtitle>
  <v-card-text class="text--primary">
    <v-row align="center">
      <v-col id="minimum-temperature"><i class="fas fa-thermometer-empty temp-font"></i>
        Min: 21.1°
      </v-col>
      <v-col id="maximum-temperature"><i class="fas fa-thermometer-full temp-font"></i>
        Max: 32.2°
      </v-col>
    </v-row>
    <v-row align="center">
      <v-col>
        <v-tabs icons-and-text="">
          <v-tab>
            Day
            <v-icon>wb_sunny</v-icon>
          </v-tab>
          <v-tab>
            Night
            <v-icon>nights_stay</v-icon>
          </v-tab>
          <v-tab>
            Flight
            <v-icon>airplanemode_active</v-icon>
          </v-tab>
          <v-tabs-items>
            <v-tab-item>
              <!---->
              <div>
                <v-row>
                  <v-col cols="6">
                    <v-img id="weather-day-con" src="/accuweathericons/6-s.png"></v-img>
                  </v-col>
                  <v-col cols="6">
                    <v-icon>toys</v-icon> <label id="weather-day-wind-speed">Wind : 18.8 </label>
                    <hr>
                    <v-icon>invert_colors</v-icon> <label id="weather-day-rain-speed">Rain : 23.1/ </label>
                    <hr>
                    <v-icon>ac_unit</v-icon> <label id="weather-day-snow">Snow : 10.1/ </label>
                    <hr>
                  </v-col>
                </v-row>
              </div>
              <v-row>
                <v-col>
                  <h2></h2> <span></span>
                </v-col>
              </v-row>
            </v-tab-item>
            <v-tab-item>
              <!---->
              <div>
                <v-row>
                  <v-col cols="6">
                    <v-img src="/accuweathericons/6-s.png"></v-img>
                  </v-col>
                  <v-col cols="6">
                    <v-icon>toys</v-icon> <label id="weather-night-wind-speed">Wind : 15 </label>
                    <hr>
                    <v-icon>invert_colors</v-icon> <label id="weather-night-rain-speed">Rain : 56/ </label>
                    <hr>
                    <v-icon>ac_unit</v-icon> <label id="weather-night-snow">Snow : 10.2/ </label>
                    <hr>
                  </v-col>
                </v-row>
                <v-row>
                  <v-col>
                    <h2></h2> <span></span>
                  </v-col>
                </v-row>
              </div>
            </v-tab-item>
            <v-tab-item>
              <!---->
              <div>
                <v-row>
                  <v-col cols="4">
                    <v-icon>airplanemode_active</v-icon>
                  </v-col>
                  <v-col cols="8"><label id="flight-source-destination">India - Amsterdam</label>
                    <hr> <label id="flight-price">Price : EUR 490</label>
                    <hr> <label id="flight-distance">Distance : 6584.75</label>
                    <hr> <label id="flight-duration">Duration : 43h 20m</label>
                    <hr> <label id="flight-stops">Stops : 0</label>
                    <hr> <label id="flight-seats">Seats available : 7</label>
                    <hr>
                  </v-col>
                </v-row>
              </div>
            </v-tab-item>
            <v-row>
              <v-btn type="submit" color="warning">
                Select
              </v-btn>
            </v-row>
          </v-tabs-items>
        </v-tabs>
      </v-col>
    </v-row>
  </v-card-text>
</v-card>
`;
